src/time_integration/methods_IMEXJinXin.jl

```diff
@@ -0,0 +1,728 @@
+    if length(integrator.opts.tstops) > 1
+        pop!(integrator.opts.tstops)
+    end
+    push!(integrator.opts.tstops, integrator.tdir * t)
+end
+
+has_tstop(integrator::SimpleIntegratorIMEX) = !isempty(integrator.opts.tstops)
+first_tstop(integrator::SimpleIntegratorIMEX) = first(integrator.opts.tstops)
+
+# Forward integrator.stats.naccept to integrator.iter (see GitHub PR#771)
+function Base.getproperty(integrator::SimpleIntegratorIMEX, field::Symbol)
+    if field === :stats
+        return (naccept = getfield(integrator, :iter),)
+    end
+    # general fallback
+    return getfield(integrator, field)
+end
+
+"""
+    solve(ode, alg; dt, callbacks, kwargs...)
+
+The following structures and methods provide the infrastructure for SSP Runge-Kutta methods
+of type `SimpleAlgorithmSSP`.
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+"""
+function solve(ode::ODEProblem, alg::SimpleAlgorithmIMEX;
+               dt, callback = nothing, kwargs...)
+    u = copy(ode.u0)
+    du1 = similar(u)
+    du2 = similar(u)
+    du3 = similar(u)
+    fu1 = similar(u)
+    fu2 = similar(u)
+    fu3 = similar(u)
+    du = similar(u)
+    u1 = similar(u)
+    u2 = similar(u)
+    u3 = similar(u)
+    u_tmp1 = similar(u)
+    u_tmp2 = similar(u)
+    u_tmp3 = similar(u)
+    r0 = similar(u)
+    t = first(ode.tspan)
+    tdir = sign(ode.tspan[end] - ode.tspan[1])
+    iter = 0
+    integrator = SimpleIntegratorIMEX(u, u1, u2, u3, fu1, fu2, fu3, du1, du2, du3, du,
+                                      u_tmp1, u_tmp2, u_tmp3, r0, t, tdir, dt, dt, iter,
+                                      ode.p,
+                                      (prob = ode,), ode.f, alg,
+                                      SimpleIntegratorIMEXOptions(callback, ode.tspan;
+                                                                  kwargs...),
+                                      false, true, false)
+
+    # resize container
+    resize!(integrator.p, nelements(integrator.p.solver, integrator.p.cache))
+
+    # initialize callbacks
+    if callback isa CallbackSet
+        foreach(callback.continuous_callbacks) do cb
+            error("unsupported")
+        end
+        foreach(callback.discrete_callbacks) do cb
+            cb.initialize(cb, integrator.u, integrator.t, integrator)
+        end
+    elseif !isnothing(callback)
+        error("unsupported")
+    end
+
+    for stage_callback in alg.stage_callbacks
+        init_callback(stage_callback, integrator.p)
+    end
+
+    solve!(integrator)
+end
+
+function solve!(integrator::SimpleIntegratorIMEX)
+    @unpack prob = integrator.sol
+    @unpack alg = integrator
+    t_end = last(prob.tspan)
+    callbacks = integrator.opts.callback
+    semi = integrator.p
+    mesh, equations, solver, cache = Trixi.mesh_equations_solver_cache(semi)
+    relaxation_rate = equations.eps_relaxation
```

Comment:
Ideally, we should have two options for this type of integrator:

- If Jin-Xin relaxation equations are used, do the IMEX stuff
- Otherwise, just apply the explicit method

We may not need to implement this right now but should keep a TODO note